packages/@vuepress/theme-default/src/shared/options.ts

```diff
@@ -28,7 +29,7 @@ export interface DefaultThemePluginsOptions {
   /**
    * Enable @vuepress/plugin-external-link-icon or not
    */
-  externalLinkIcon?: boolean
+  externalLinkIcon?: ExternalLinkIconPluginOptions | boolean
```

Comment:
I'd prefer to keep `themePlugins` consistent, i.e. keep 	`externalLinkIcon?: boolean` here.

We can still keep the `openInNewWindow` locale config, and use a `resolveExternalLinkIconPluginOptions` util to resolve the options.